### ISSUE TITLE
Support for USB through the Orange Pi Zero Extension board

### ISSUE THREAD

**Issue by ykasidit** [nguiard/archlinux-orange-pi-zero#6]
Hello,

Thanks for this image!

Just like to report and request some guidance on how to get the USB on extension board working - right now lsusb doesnt show them - same issue as for this solution for Armbian in this video: https://www.youtube.com/watch?v=9iJUUxNyGR4 but cant use that solution and dont know how to apply it to this arch image - please guide.

**Comment by nguiard** [comment]
Hello,
Sorry for the late answer. I don't know how to fix this right now or how to apply the fix appearing in the video. It might be that updating the kernel to the latest version would fix the issue.
My image has the kernel locked to an old version, because at the time I built it, versions after that one would break ethernet support. However, since then the kernel does support ethernet alright on this board. If by chance you still have the kernel locked, you can unlock it by commenting the line regarding the linux kernel in /etc/pacman.conf.
If you try that, please report if it worked. I unfortunately can't investigate more for now.

**Comment by yurilukin** [comment]
Hello,
I tried your image and can confirm that USB ports on extension board does not works. But I found solution - patched file sun8i-h2-plus-orangepi-zero.dtb on russian forum 4pda.ru. I just put it to /boot/dtbs and all USB ports now works!
Link to download patched sun8i-h2-plus-orangepi-zero.dtb: [https://yadi.sk/d/ETdfV3N_6Be2cA)](url)

Thank you for your image, it works like a charm.

**Comment by nguiard** [comment]
Hello,
Thanks yurilukin for your feedback! I have updated the image and the README to include an explanation about the USB issue on the expansion board. Also, the file sun8i-h2-plus-orangepi-zero-3usb.dtb you have linked to is now included in the image for convenience (but is not installed by default) for those who may need it.

**Comment by RemcoFYM** [comment]
Here the two extra USB ports are also not recognized, nor after the instruction video (see link above). Is there a cookbook recipe describing exactly what to do in order to let my OPi Zero detect these USB ports? 

I run (uname -a) : 
Linux orangepizero 5.3.9-sunxi #19.11.3 SMP Mon Nov 18 18:49:43 CET 2019 armv7l GNU/Linux

**Comment by yurilukin** [comment]
Nguiard already added "USB on Orange Pi expansion board" section to README.md.
Step by step in simple:
1. IMPORTANT! Backup or rename old sun8i-h2-plus-orangepi-zero.dtb in /boot/dtbs directory. In the case of any problems you will always can restore it.
2. Put the patched sun8i-h2-plus-orangepi-zero-3usb.dtb in /boot/dtbs (if it is not there).
3. Rename it to sun8i-h2-plus-orangepi-zero.dtb
4. Reboot.

**Comment by RemcoFYM** [comment]
OK all understood, but where can I fetch/get sun8i-h2-plus-orangepi-zero-3usb.dtb ?

Edit: I don't have /boot/dtbs, only /boot/dtb (without an 's')

**Comment by yurilukin** [comment]
File is [here](https://drive.google.com/open?id=1k3G3yB-7Et2ByGTLvagEGhwHBu0YF-tk).

It is very strange that directory name is "dtb". Try to locate file sun8i-h2-plus-orangepi-zero.dtb in this directory or in whole filesystem (commands "find" and "locate").

**Comment by RemcoFYM** [comment]
Thanks, no my dir name is 'dtb' , see: http://dpaste.com/17RZ6E2




On Friday 03 January 2020 07:54:27 yurilukin wrote:

File is here.
It is very strange that directory name is "dtb". Try to locate file sun8i-h2-
plus-orangepi-zero.dtb in this directory or in whole filesystem (commands 
"find" and "locate").

**Comment by RemcoFYM** [comment]
Thanks for the reaction. I gave you a reply where I gave an ls -al of my /boot directory : see, 
http://dpaste.com/17RZ6E2

No dtbs here .. (?)

**Comment by yurilukin** [comment]
I see armbian* files in your /boot. Are you sure that you have installed Arch? If it is Armbian you should fix problem by ["armbian way"](https://forum.armbian.com/topic/3718-orange-pi-zero-dev-kernel-does-not-see-expansion-usb-ports/) (adding "overlays=usbhost2 usbhost3" to /boot/armbianEnv.txt).

**Comment by RemcoFYM** [comment]
Yes, I use armbian. Sorry, didn't knew there were other flavours. 

Orange Pi Zero = Orange Pi Zero, sorry for the misunderstanding. 

I tried the overlays in armbianEnv.txt, but no positive result here. 

Thanks for your efforts!

On Saturday 04 January 2020 00:46:27 yurilukin wrote:

I see armbian* files in your /boot. Are you sure that you have installed Arch? 
If it is Armbian you should fix problem by "armbian way" (adding 
"overlays=usbhost2 usbhost3" to /boot/armbianEnv.txt).

**Comment by RemcoFYM** [comment]
Found the root cause of my issue: corrupted armbianEnv.txt file (!!)

See: https://forum.armbian.com/topic/7855-corrupted-armbianenvtxt-in-opz-
fails-to-access-expansion-board/

I mounted the original Armbian image and copied that armbianEnv.txt
into the OPi Zero's /boot, rebooted .. now all is working : -)

On Saturday 04 January 2020 00:46:27 yurilukin wrote:

I see armbian* files in your /boot. Are you sure that you have installed Arch? 
If it is Armbian you should fix problem by "armbian way" (adding 
"overlays=usbhost2 usbhost3" to /boot/armbianEnv.txt).
